Sources/WalletConnectRelay/RelayClient.swift

```diff
@@ -194,48 +218,38 @@ public final class RelayClient {
                 completion(nil)
             }
         }
-        cancellable = requestAcknowledgePublisher
-            .filter {$0.id == request.id}
-            .sink { (_) in
-                cancellable?.cancel()
-                completion(nil)
-            }
-        return request.id
-    }
-
-    private func setUpBindings() {
-        dispatcher.onMessage = { [weak self] payload in
-            self?.handlePayloadMessage(payload)
-        }
-        dispatcher.onConnect = { [unowned self] in
-            self.socketConnectionStatusPublisherSubject.send(.connected)
-        }
     }
 
+    // FIXME: Parse data to string once before trying to decode -> respond error on fail
     private func handlePayloadMessage(_ payload: String) {
-        if let request = tryDecode(SubscriptionRequest.self, from: payload), validate(request: request, method: .subscription) {
-            do {
-                try jsonRpcSubscriptionsHistory.set(topic: request.params.data.topic, request: request)
-                onMessage?(request.params.data.topic, request.params.data.message)
-                acknowledgeSubscription(requestId: request.id)
-            } catch {
-                logger.info("Relay Client Info: Json Rpc Duplicate Detected")
+        if let request = tryDecode(RPCRequest.self, from: payload) {
+            if let params = try? request.params?.get(Subscription.Params.self) {
+                do {
+                    try rpcHistory.set(request, forTopic: params.data.topic, emmitedBy: .remote)
+                    try acknowledgeRequest(request)
+                    onMessage?(params.data.topic, params.data.message)
+                } catch {
+                    logger.error("[RelayClient] RPC History 'set()' error: \(error)")
+                }
+            } else {
+                logger.error("Unexpected request from network")
+            }
+        } else if let response = tryDecode(RPCResponse.self, from: payload) {
+            switch response.outcome {
+            case .success(let anyCodable):
+                if let _ = try? anyCodable.get(Bool.self) { // TODO: Handle success vs. error
+                    requestAcknowledgePublisherSubject.send(response.id)
+                } else if let subscriptionId = try? anyCodable.get(String.self) {
+                    subscriptionResponsePublisherSubject.send((response.id, subscriptionId))
+                }
+            case .failure(let rpcError):
+                logger.error("Received error message from iridium network, code: \(rpcError.code), message: \(rpcError.message)")
```

Comment:
maybe we could use `relay network` instead of `iridium network`? as it is suppose to change.

Reply:
True, I've removed references to the "iridium" naming